js/dataframe/Index.svelte

```diff
@@ -46,75 +46,18 @@
 		display: boolean;
 	}[];
 	export let max_height: number | undefined = undefined;
-
 	export let loading_status: LoadingStatus;
 	export let interactive: boolean;
 
-	let _headers: Headers;
-	let display_value: string[][] | null;
-	let styling: string[][] | null;
-	let values: (string | number)[][];
-	async function handle_change(data?: {
-		data: Data;
-		headers: Headers;
-		metadata: Metadata;
-	}): Promise<void> {
-		let _data = data || value;
-
-		_headers = [...(_data.headers || headers)];
-		values = _data.data ? [..._data.data] : [];
-		display_value = _data?.metadata?.display_value
-			? [..._data?.metadata?.display_value]
+	$: _headers = [...(value.headers || headers)];
+	$: cell_values = value.data ? [...value.data] : [];
+	$: display_value = value?.metadata?.display_value
+		? [...value?.metadata?.display_value]
+		: null;
+	$: styling =
+		!interactive && value?.metadata?.styling
+			? [...value?.metadata?.styling]
 			: null;
-		styling =
-			!interactive && _data?.metadata?.styling
-				? [..._data?.metadata?.styling]
-				: null;
-		await tick();
-
-		gradio.dispatch("change");
-		if (!value_is_output) {
-			gradio.dispatch("input");
-		}
-	}
-
-	handle_change();
-
-	afterUpdate(() => {
-		value_is_output = false;
-	});
-
-	$: {
-		if (old_value && JSON.stringify(value) !== old_value) {
-			old_value = JSON.stringify(value);
-			handle_change();
-		}
-	}
-
-	if (
-		(Array.isArray(value) && value?.[0]?.length === 0) ||
-		value.data?.[0]?.length === 0
-	) {
-		value = {
-			data: [Array(col_count?.[0] || 3).fill("")],
-			headers: Array(col_count?.[0] || 3)
-				.fill("")
-				.map((_, i) => `${i + 1}`),
-			metadata: null
-		};
-	}
```

Comment:
Moved to the backend `def _is_empty()` to avoid duplicated logic related to `self.empty_input`